### PR TITLE
Detections and hunting queries- Aug 2020

### PR DESCRIPTION
**Pull request by ashwin-patil**
Fixes #

## Proposed Changes

  - Hunting query for Suspicious Login outside normal hours
  - Detection for Expensive Computes based on Azure Activity logs

### CODE REVIEW

File: Detections/AzureActivity/Creation_of_Expensive_Computes.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+id: 9736e5f1-7b6e-4bfb-a708-e53ff1d182c3
+name: Creation of expensive computes in Azure
+description: |
+  'Looks for creation of large size/expensive VMs (GPU or with large no of virtual CPUs) in Azure.
```
Comment (shainw) [inline Detections/AzureActivity/Creation_of_Expensive_Computes.yaml:4]: Identifies the creation of...

File: Detections/AzureActivity/Creation_of_Expensive_Computes.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+id: 9736e5f1-7b6e-4bfb-a708-e53ff1d182c3
+name: Creation of expensive computes in Azure
```
Comment (shainw) [inline Detections/AzureActivity/Creation_of_Expensive_Computes.yaml:2]: Just go with the full name in the file name or change the name here to `Creation of expensive computes` to match the file name.

File: Detections/AzureActivity/Creation_of_Expensive_Computes.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  Adversary may create new or update existing virtual machines sizes to evade defenses 
+  or use it for cryptomining purposes.
+  For Windows/Linux Vm Sizes - https://docs.microsoft.com/azure/virtual-machines/windows/sizes 
+  Azure VM Naming Conventions - https://docs.microsoft.com/azure/virtual-machines/vm-naming-conventions'
+requiredDataConnectors:
+  - connectorId: AzureActivity
+    dataTypes:
+      - AzureActivity
+queryFrequency: 1d
+queryPeriod: 14d
```
Comment (shainw) [inline Detections/AzureActivity/Creation_of_Expensive_Computes.yaml:14]: 1d

File: Hunting Queries/SecurityEvent/Suspicious_Login_outside_normal_hours.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
+id: e7bfbc3f-98c7-4aaa-a64c-de9c058b86b2
+name: Suspicious Login outside normal hours
```
Comment (shainw) [inline Hunting Queries/SecurityEvent/Suspicious_Login_outside_normal_hours.yaml:2]: Lets title this with Windows so it is clear where it is coming from, also this means a file name change to include Windows.

File: Hunting Queries/SecurityEvent/Suspicious_Login_outside_normal_hours.yaml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      AllLogonEvents
+      | where StartTime > ago(lookback)
+  )
+  on TargetUserName
+  // Filtering for logon events based on range of max and min of historical logon hour values seen
+  | where HourOfLogin > max_HourOfLogin or HourOfLogin < min_HourOfLogin
+  // Also populating additional column showing historical days of week when logon was seen
+  | extend historical_DayofWeek = tostring(historical_DayofWeek)
+  | summarize Total= count(), max(HourOfLogin), min(HourOfLogin), current_DayofWeek =make_set(DayofWeek), StartTime=max(StartTime), EndTime = min(StartTime), SourceIP = make_set(IpAddress), SourceHost = make_set(WorkstationName), SubjectUserName = make_set(SubjectUserName), HostLoggedOn = make_set(Computer) by EventID, Activity, TargetDomainName, TargetUserName , ProcessName , LogonTypeName, StatusDesc, SubStatusDesc, historical_DayofWeek
+  | extend historical_DayofWeek = todynamic(historical_DayofWeek)
```
Comment (shainw) [inline Hunting Queries/SecurityEvent/Suspicious_Login_outside_normal_hours.yaml:107]: Need the detection required parameters - | extend timestamp = StartTime, AccountCustomEntity = TargetUserName

**Comment by ashwin-patil** [comment]
Made the changes per PR review feedback

**Review by shainw — APPROVED** [review]

**Pull request merged**